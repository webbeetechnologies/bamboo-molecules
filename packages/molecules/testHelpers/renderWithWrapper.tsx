import type { ReactElement, JSXElementConstructor, ReactNode } from 'react';
import { render, RenderOptions } from '@testing-library/react-native';
import { ProvideMolecules } from '../src';

export const renderWithWrapper = <P extends {} = {}>(
    component: ReactElement<P, string | JSXElementConstructor<P>>,
    options?: RenderOptions,
): ReturnType<typeof render> => {
    const Wrapper = ({ children }: { children: ReactNode }) => (
        <ProvideMolecules>{children}</ProvideMolecules>
    );

    return render(component, { wrapper: Wrapper, ...options });
};
