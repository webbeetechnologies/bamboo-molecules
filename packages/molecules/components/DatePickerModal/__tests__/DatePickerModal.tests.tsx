import { renderWithWrapper } from '../../../../testHelpers';
import { DatePickerModal } from '../index';

const onChangeMock = jest.fn();
const onDismissMock = jest.fn();
const onConfirmMock = jest.fn();

describe('DatePickerModal', () => {
    it('renders DatePickerModal component with range mode', () => {
        const tree = renderWithWrapper(
            <DatePickerModal
                isOpen={false}
                mode="range"
                startDate={new Date()}
                endDate={undefined}
                onChange={onChangeMock}
                onClose={onDismissMock}
                onConfirm={onConfirmMock}
            />,
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
