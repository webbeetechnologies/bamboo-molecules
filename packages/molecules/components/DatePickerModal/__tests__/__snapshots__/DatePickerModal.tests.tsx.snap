// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePickerModal renders DatePickerModal component with range mode 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[MockFunction]}
    presentationStyle="overFullScreen"
    statusBarTranslucent={true}
    supportedOrientations={
      Array [
        "portrait",
        "portrait-upside-down",
        "landscape",
        "landscape-left",
        "landscape-right",
      ]
    }
    transparent={true}
    visible={false}
  />
</View>
`;
